app/controllers/admin/cabinet_ministers_controller.rb

```diff
@@ -68,4 +68,8 @@ def order_ministerial_roles_params
   def order_ministerial_organisations_params
     params.require(:ministerial_organisations)["ordering"]
   end
+
+  def ministerial_roles_with_current_appointments
+    MinisterialRole.includes(:translations).where(cabinet_member: true).order(:seniority).select { |role| role.current_role_appointments.any? }
+  end
```

Comment:
This should probably move belong on the `Ministerial` role class. Might be nice to have a `cabinet_members` scope as well

Reply:
Also is this creating an N+1 query by checking the current role appointments for each role? We should probably just inner join to the role_appointments table which will naturally filter out any ministerial roles without appointments